lib/core/versions/0.5.0/OperationProcessor.ts

```diff
@@ -20,81 +20,77 @@ export default class OperationProcessor implements IOperationProcessor {
     anchoredOperationModel: AnchoredOperationModel,
     didDocumentReference: { didDocument: object | undefined }
   ): Promise<ApplyResult> {
-    let operationHash = undefined;
-
     try {
       // NOTE: only used for read interally.
       const didDocument = didDocumentReference.didDocument as (DocumentModel | undefined);
-
       const operation = AnchoredOperation.createAnchoredOperation(anchoredOperationModel);
-      operationHash = operation.operationHash;
 
       if (operation.type === OperationType.Create) {
 
         // If we have seen a previous create operation.
         if (didDocumentReference.didDocument) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         const originalDidDocument = Document.from(operation.encodedPayload, this.didMethodName, ProtocolParameters.hashAlgorithmInMultihashCode)!;
 
         const signingKey = Document.getPublicKey(originalDidDocument, operation.signingKeyId);
 
         if (!signingKey) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         if (!(await operation.verifySignature(signingKey))) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         didDocumentReference.didDocument = originalDidDocument;
-        return { validOperation: true, operationHash };
+        return { validOperation: true };
       } else if (operation.type === OperationType.Delete) {
         // Delete can be applied only on valid did with a current document
         if (!didDocument) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         // The current did document should contain the public key mentioned in the operation ...
         const publicKey = Document.getPublicKey(didDocument, operation.signingKeyId);
         if (!publicKey) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         // ... and the signature should verify
         if (!(await operation.verifySignature(publicKey))) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
         }
 
         // If the delete is valid
         didDocumentReference.didDocument = undefined;
-        return { validOperation: true, operationHash };
+        return { validOperation: true };
       } else {
         // Update operation
 
         // If we have not seen a valid create operation yet.
         if (didDocument === undefined) {
-          return { validOperation: false, operationHash };
+          return { validOperation: false };
```

Comment:
These are repeated a lot. Would it make sense to have a default return object and just return it on these conditions? Then in the validOperation = true case, manipulate the default object so we can change all of them by changing just one place.

Reply:
I thought the same, a constant for each case would be more readable: 

```
const InvalidOperationResponse = { validOperation: false };
...
return InvalidOperationResponse;
```

readability isn't that bad now, but if the response type starts to get more complicated I could see this helping...

Reply:
I think this is the same point that Orie pointed out earlier, which is totally fair. But as explained earlier I intend to won't fix this because:

1. This is old versioned code, so don't intend to do make any changes unless really needed. Latest code (under `latest` already made improvements on this old code, feel free to check it out).
1. We are planning to reset the genesis block in the testnet in the next ION release, at which point this code will be removed forever.